src/rp2/accounting_engine.py

```diff
@@ -189,18 +206,19 @@ def get_acquired_lot_for_taxable_event(
             if avl_result.acquired_lot != self.__acquired_lot_list[avl_result.index]:
                 raise RP2RuntimeError("Internal error: acquired_lot incongruence in accounting logic")
             method = self._get_accounting_method(taxable_event.timestamp.year)
-            lot_candidates: AcquiredLotCandidates = AcquiredLotCandidates(
-                method, self.__acquired_lot_list, self.__acquired_lot_2_partial_amount, avl_result.index
-            )
-            acquired_lot_and_amount: Optional[AcquiredLotAndAmount] = method.seek_non_exhausted_acquired_lot(
-                lot_candidates, taxable_event, new_taxable_event_amount
-            )
-            if acquired_lot_and_amount:
-                return TaxableEventAndAcquiredLot(
-                    taxable_event=taxable_event,
-                    acquired_lot=acquired_lot_and_amount.acquired_lot,
-                    taxable_event_amount=new_taxable_event_amount,
-                    acquired_lot_amount=acquired_lot_and_amount.amount,
+            lot_candidates: Optional[AcquiredLotCandidates] = self.__years_2_lot_candidates.find_max_value_less_than(taxable_event.timestamp.year)
+
+            if lot_candidates:
```

Comment:
You're right but the type system doesn't know that - the annotation comes from here: https://github.com/eprbell/prezzemolo/blob/main/src/prezzemolo/avl_tree.py#L111

So we can cast to non-`Optional`, `assert` lot_candidates is not None`, or have an `if` that is always True.

Reply:
Ah, right: in this case I have a slight preference for an assert, because it makes it clear to the reader that None shouldn't happen and it avoids an extra level of indentation.

Reply:
Done

Reply:
@eprbell Looks like `bandit` isn't happy with the assert - what's your preference on how to resolve?

Reply:
OK, in that case let's restore the original code with the `if` statement. Maybe add a small comment saying that None is not expected as a value.

Reply:
Updated, tests are passing